python/setup.py

```diff
@@ -49,64 +54,76 @@
+        shutil.rmtree(setup_file_path + '/__pycache__', ignore_errors=True)
+
+        clean_folder(setup_file_path + '/cuml')
+
+    except IOError:
+        pass
+
+    # need to terminate script so cythonizing doesn't get triggered after
+    # cleanup unintendedly
+    sys.argv.remove("clean")
+    sys.argv.remove("--all")
+
+    if len(sys.argv) == 1:
+        sys.exit(0)
 
 ##############################################################################
-# - Subrepo checking and cloning ---------------------------------------------
+# - Cloning dependencies if needed -------------------------------------------
 
 subrepos = [
-    'cub',
-    'cutlass',
-    'faiss',
     'treelite'
 ]
 
 # We check if there is a libcuml++ build folder, by default in cpp/build
 # or in CUML_BUILD_PATH env variable. Otherwise setup.py will clone the
-# dependencies defined in cpp/CMakeListst.txt
-if "clean" not in sys.argv:
-    if os.environ.get('CUML_BUILD_PATH', False):
-        libcuml_path = '../' + os.environ.get('CUML_BUILD_PATH')
-    else:
-        libcuml_path = '../cpp/build/'
+# dependencies defined in cpp/cmake/Dependencies.cmake
+if os.environ.get('CUML_BUILD_PATH', False):
+    libcuml_path = '../' + os.environ.get('CUML_BUILD_PATH')
+else:
+    libcuml_path = '../cpp/build/'
 
-    found_cmake_repos = get_submodule_dependencies(subrepos,
-                                                   libcuml_path=libcuml_path)
-
-    if found_cmake_repos:
-        treelite_path = os.path.join(libcuml_path,
-                                     'treelite/src/treelite/include')
-        faiss_path = os.path.join(libcuml_path, 'faiss/src/')
-        cub_path = os.path.join(libcuml_path, 'cub/src/cub')
-        cutlass_path = os.path.join(libcuml_path, 'cutlass/src/cutlass')
-    else:
-        # faiss requires the include to be to the parent of the root of
-        # their repo instead of the full path like the others
-        faiss_path = 'external_repositories/'
-        treelite_path = 'external_repositories/treelite/include'
-        cub_path = 'external_repositories/cub'
-        cutlass_path = 'external_repositories/cutlass'
+found_cmake_repos = get_submodule_dependencies(subrepos,
+                                               libcuml_path=libcuml_path)
 
+if found_cmake_repos:
+    treelite_path = os.path.join(libcuml_path,
+                                 'treelite/src/treelite/include')
 else:
-    subprocess.check_call(['rm', '-rf', 'external_repositories'])
-    treelite_path = ""
-    faiss_path = ""
-    cub_path = ""
-    cutlass_path = ""
+    treelite_path = 'external_repositories/treelite/include'
+
 
 ##############################################################################
 # - Cython extensions build and parameters -----------------------------------
 
+# cumlcomms and nccl are still needed for multigpu algos not based
+# on libcumlprims
 libs = ['cuda',
         'cuml++',
+        'cumlcomms',
+        'nccl',
```

Comment:
not sure I get this @dantegd . `--singlegpu` is to build only singlegpu algos, right? So, why do we `cumlcomms` and `nccl` for this case?

Reply:
`--singlegpu` removes the algorithms that use `libcumlprims`, so it doesn't remove things like cuml.dask RF and NearestNeighbors for example. I don't think we've had a scenario where users would be interested in a cuML without those (as opposed to platforms where `libcumlprims` is not available), so I don't think its necessary. The main thing avoided would be requiring `nccl` for single gpu algos, so that could be a follow up PR if you think that would be good to have (cc @JohnZed for additional input if that would be a desired behavior of `--singlegpu`)

Reply:
In my mind, `--singlegpu` has always meant no comms (and thus no nccl) as well as no libcumlprims. Please correct if this is wrong.

Reply:
Not really, it was introduced at the same time as libcumlprims and so far it’s main purpose has been to build where libcumlprims is not available. We could make it so that it fully avoids using nccl, but that is a different work beyond the scope of this PR: having a comprehensive python clean target (and is needed to make —singlegpu to work for now)

Reply:
Got it. Thanks Dante.